### PR TITLE
Use mozilla-actions/sccache-action to speed up release and nightly release workflow

### PR DESCRIPTION
**Pull request by hustcer**
# Description

I have tried `mozilla-actions/sccache-action` in the `nushell/nightly` repo for a while, and the workflow running logs could be found here: https://github.com/nushell/nightly/actions/runs/6607933986/attempts/1, the workflow has been run 5 times:
- [Attempt #1](https://github.com/nushell/nightly/actions/runs/6607933986/attempts/1) cost 46min
- [Attempt #2](https://github.com/nushell/nightly/actions/runs/6607933986/attempts/2) cost 34min 
- [Attempt #3](https://github.com/nushell/nightly/actions/runs/6607933986/attempts/3) cost 35min 
- [Attempt #4](https://github.com/nushell/nightly/actions/runs/6607933986/attempts/4) cost 23min
- [Attempt #5](https://github.com/nushell/nightly/actions/runs/6607933986) cost 29min

Some explanations: 

1. `Attempt #1` cost the longest time, as it's the first run and with nothing been cached.
2. All the following running of workflow taken an obvious shorter time, Reduced by 20% to 50%
3. All the binary release time for Linux and macOS have been reduced significantly, however it's not that useful for Windows.

In summary, `mozilla-actions/sccache-action` do improve the release time of workflow. We can give it a try, and the config is quite simple BTW.

### CODE REVIEW

**Comment by fdncred** [comment]
Nice work @hustcer. The question is will sscache improve over rust-cache that we currently use. Were you using rust-cache in the nightly repo before trying sscache?

**Comment by hustcer** [comment]
> The question is will sscache improve over rust-cache that we currently use. Were you using rust-cache in the nightly repo before trying sscache?

We use `actions-rust-lang/setup-rust-toolchain` in release and nightly release workflow, and the cache is enabled by default: https://github.com/actions-rust-lang/setup-rust-toolchain#inputs

**Comment by sholderbach** [comment]
Conceptual question: Should we use caching for releases? With the release the priority is that the build is correct and optimized in the best way possible. Could this caching have a negative impact on that?


I absolutely want fast builds for the regular CI.

**Comment by fdncred** [comment]
I'm also curious to see what change this would make in ci.yml.
